docs/appendix/jupyterlab_extensions.rst

```diff
@@ -0,0 +1,181 @@
+
+* ``assignmentlist.ts``
+
+	* Contains all the logic necessary to display the assignments.
+
+* ``handlers.py``
+
+	* Defines the backend of the extension.
+	* Uses the nbgrader ExchangeList, ExchangeFetchAssignment, ExchagneFetchFeedback, and ExchangeSubmit classes.  
+
+Classes
+"""""""
+AssignmentList
+	* Used to load and display the list of released, downloaded, and submitted assignments. 
+
+Assignment
+	* Creates the rows for each assignment. Each row consists of a link, a span element to display the name of the course, and a button.
+
+Submission
+	* Makes a submission row which consists of the timestamp and a link to a feedback file if there is any.
+
+Notebook
+	* Creates a row for each notebook in an assignment. The name of the notebook is a link to open the notebook and each row also contains a button to validate the notebook (run the tests for the notebook).
+
+CourseList
+	* Used to load and display the  course dropdown. 
+	* When you click on a course it switches to to display the assignments for that course.
+
+.. image:: ../extension_screenshots/assignment_list.png
+   :width: 600
+
+Create Assignment
+-----------------
+In Jupyter Notebooks, the extension put the UI in the cell toolbars. JupyterLab does not have cell toolbars, so we had to decide where to put the interface. We decided on a side panel which shows the nbgrader assignment information for the active notebook.
+
+What’s Done
+^^^^^^^^^^^
+* Everything (extension, styling, tests, etc.)
+
+What’s Not Done
+^^^^^^^^^^^^^^^
+* Nothing
+
+Code
+^^^^
+Files
+"""""
+* ``index.ts``
+
+	* attaches the UI to a side panel.
+* ``extension.ts``
+
+	* contains the UI elements.
+* ``model.ts``
+
+	* contains the logic which acts as an intermediary between the UI and the notebook cell metadata.
+
+Classes
+"""""""
+CreateAssignmentWidget
+	* A container for the UI, which can theoretically be attached to any widget, not just a side panel
+	* Listens to determine which notebook is the current notebook
+	* Shows the NotebookWidget for the current notebook
+
+NotebookWidget
+	* Contains the UI associated with a notebook
+	* Has a NotebookHeaderWidget at the top and a NotebookPanelWidget which takes up the remaining space
+NotebookHeaderWidget
+	* Currently, only contains the total points for the assignment
+NotebookPanelWidget
+	* Contains a list of CellWidgets to show the assignment information for each cell
+	* Listens to changes in the notebook cell list
+	* Adds, removes, reorders, or highlights CellWidgets to synchronize with the notebook
+CellWidget
+	* Contains the UI showing the nbgrader assignment information for one cell
+	* Reads and writes nbgrader data in the cell metadata
+
+.. image:: ../extension_screenshots/create_assignment.png
+   :width: 600
+
+Course List
+-----------
+Same functionality and layout as the course list nbextension. After installation it can be launched by opening the command palette on the left side and searching for Course List
```

Comment:
```suggestion
Same functionality and layout as the course list nbextension. After installation, it can be launched by opening the command palette on the left side and searching for Course List.
```